models/demos/resnet/tt/metalResnetBlock50.py

```diff
@@ -1401,40 +1424,59 @@ def __init__(
             ] = hardcoded_conv_blocking_and_parallelization_config[batch_size][
                 (first_conv_output_padded_nhw_size, first_conv_output_channels)
             ]
-            sliding_window_op_params = [
-                (1, 1),
-                (0, 0),
-                (4, 4),
-                (batch_size, 115, 115),
-                grid_size,
-                self.first_conv_num_cores_nhw,
-            ]
+
+            sliding_window_op_params = SlidingWindowOpParamsWithParallelConfig(
+                stride_h=1,
+                stride_w=1,
+                pad_h=0,
+                pad_w=0,
+                window_h=4,
+                window_w=4,
+                batch_size=batch_size,
+                input_h=115,
+                input_w=115,
+                num_cores_h=grid_size[1],
+                num_cores_w=grid_size[0],
+                num_cores_nhw=self.first_conv_num_cores_nhw,
```

Comment:
what does `num_cores_nhw` do?

Reply:
its mainly used to determine sharding is 1d or 2d